src/Raven.Studio/typescript/viewmodels/database/settings/databaseSettings.ts

```diff
@@ -0,0 +1,454 @@
+        
+        return $.when<pagedResult<models.settingsEntry>>({
+            items: entriesForSummaryMode,
+            totalResultCount: entriesForSummaryMode.length,
+            resultEtag: null,
+            additionalResultInfo: undefined
+        })
+    }
+
+    private fetchDatabaseSettings(db: database, refresh: boolean = false): JQueryPromise<Raven.Server.Config.SettingsResult> {
+        eventsCollector.default.reportEvent("database-settings", "get");
+
+        return new getDatabaseSettingsCommand(db, refresh)
+            .execute()
+            .done((result: Raven.Server.Config.SettingsResult) => {
+                
+                const settingsEntries = result.Settings.map(x => {
+                    const rawEntry = x as Raven.Server.Config.ConfigurationEntryDatabaseValue;
+                    
+                    if (rawEntry.Metadata.Scope === "ServerWideOnly") {
+                        return new models.serverWideOnlyEntry(rawEntry);
+                    }
+
+                    let entry: models.databaseEntry;
+                    switch (rawEntry.Metadata.Type) {
+                        case "String":
+                            entry = new models.stringEntry(rawEntry);
+                            break;
+                        case "Path":
+                            entry = new models.pathEntry(rawEntry);
+                            break;
+                        // case "Uri": // Currently there are no URI entries in the database scope
+                        //     entry = new uriEntry(rawEntry);
+                        //     break;
+                        case "Integer":
+                            entry = new models.integerEntry(rawEntry);
+                            break;
+                        case "Double":
+                            entry = new models.doubleEntry(rawEntry);
+                            break;
+                        case "Boolean":
+                            entry = new models.booleanEntry(rawEntry);
+                            break;
+                        case "Enum":
+                            entry = new models.enumEntry(rawEntry);
+                            break;
+                        case "Time":
+                            entry = new models.timeEntry(rawEntry);
+                            break;
+                        case "Size":
+                            entry = new models.sizeEntry(rawEntry);
+                            break;
+                    }
+
+                    entry.init();
+                    return entry;
+                });
+
+                this.allEntries(_.sortBy(settingsEntries, x => x.keyName()));
+            });
+    }
+
+    private isEntriesValid() {
+        let valid = true;
+        let firstError = true;
+
+        this.allEntries().map(entry => {
+            if (this.getEntryStateClass(entry) === "customized-item-with-error") {
```

Comment:
can we avoid using css classes for that?

Reply:
leaving it, as discussed